### PR TITLE
Feature: Add select/deselect all to CheckboxList

### PR DESCRIPTION
**Pull request by awcodes**
Adds optional `canSelectAll()` method to CheckboxList form component to allow selecting and deselecting all options at once.

### CODE REVIEW

File: packages/forms/docs/03-fields.md [hunk 1]
```diff
@@ -909,6 +909,21 @@ CheckboxList::make('technologies')
 
 This method accepts the same options as the `columns()` method of the [grid](layout#grid). This allows you to responsively customize the number of columns at various breakpoints.
 
+You may also allow user to select / deselect all options at once by using the `canSelectAll()` method:
```
Comment (zepfietje) [inline packages/forms/docs/03-fields.md]:
```suggestion
You may also allow users to select/deselect all options at once using the `canSelectAll()` method:
```

File: packages/forms/resources/lang/en/components.php [hunk 2]
```diff
@@ -256,4 +256,19 @@
...
+                'label' => 'Select All',
+            ],
+
+            'deselect_all' => [
+                'label' => 'Deselect All',
+            ],
+
+        ],
+
+    ],
```
Comment (zepfietje) [inline packages/forms/resources/lang/en/components.php]:
```suggestion
    ],

```

File: packages/forms/resources/lang/en/components.php [hunk 3]
```diff
@@ -256,4 +256,19 @@
...
+
+        'buttons' => [
+
+            'select_all' => [
+                'label' => 'Select All',
+            ],
+
+            'deselect_all' => [
+                'label' => 'Deselect All',
+            ],
```
Comment (zepfietje) [inline packages/forms/resources/lang/en/components.php]:
```suggestion
            'deselect_all' => [
                'label' => 'Deselect all',
            ],

            'select_all' => [
                'label' => 'Select all',
            ],
```

**Review by zepfietje — CHANGES_REQUESTED** [review]

File: packages/forms/src/Components/CheckboxList.php [hunk 4]
```diff
@@ -89,6 +91,13 @@ public function relationship(string | Closure $relationshipName, string | Closur
         return $this;
     }
 
+    public function toggleable(bool | Closure $condition = true): static
```
Comment (danharrin) [inline packages/forms/src/Components/CheckboxList.php]:
```suggestion
    public function bulkToggleable(bool | Closure $condition = true): static
```

**Comment by zepfietje** [comment]
@danharrin, any action needed on this PR?

File: packages/forms/docs/03-fields.md [hunk 5]
```diff
@@ -909,6 +909,21 @@ CheckboxList::make('technologies')
 
 This method accepts the same options as the `columns()` method of the [grid](layout#grid). This allows you to responsively customize the number of columns at various breakpoints.
 
+You may also allow users to select/deselect all options at once using the `bulkToggleable()` method:
```
Comment (danharrin) [inline packages/forms/docs/03-fields.md]:
```suggestion
You may also allow users to toggle all checkboxes at once using the `bulkToggleable()` method:
```

**Pull request merged**